### PR TITLE
Improve styling of the calendar

### PR DESCRIPTION
**Pull request by haukkagu**
# Notable changes
- Changed design of the event entry to a pill design
- Added time and place for the event in the popover
- Events that last over multiple days will be shown on each day they take place.
- Made the calendar taller and decreased the font size for the day numbers and weekend heading.

In addition to many minor changes like centering the registration pill in the popover, improving variable names, changing colors etc.

# Result
### Example 1 (light mode):
before1
after1

### Example 2 (dark mode):
before2
after2

### Example 3 (another month with popover):
before4
after4

# Testing

- [sort of ] I have thoroughly tested my changes.
I have scrolled through the calendar and all the months look fine.

# Notes
- I mentioned that I wanted to improve the styling of the calendar on phones. That will be implemented in a separate PR.
- I realize that the Pill-component can be used for the event pills instead of implementing it myself. I remember trying to do that when I first started on this project, but couldn't get it to work. However, I may give it another shot now.

---

Resolves [this linear task](https://linear.app/abakus-webkom/issue/ABA-176/improve-the-layout-of-the-calendar)

### CODE REVIEW

File: app/routes/events/utils.ts [hunk 1]
```diff
@@ -47,6 +47,24 @@ export const colorForEvent = (eventType: EventType) => {
   return COLOR_CONSTANTS[eventType] || COLOR_CONSTANTS['other'];
 };
 
+// Hard-coded text colors for different event types
+export const TEXT_COLOR_CONSTANTS: Record<EventType, string> = {
```
Comment (ivarnakken) [inline app/routes/events/utils.ts:51]: This doesn't have to exported, or?

**Review by ivarnakken — CHANGES_REQUESTED** [review]
Looks much much better! Great work! 😍 😍 

Ideally I'd want the bars to overflow over the borders when the event stretches over multiple days, but that can be fixed later 👍🏼 

Btw, there are some linting issues as well. Please amend those changes to a previous commit.

File: app/routes/events/components/Calendar.css [hunk 2]
```diff
@@ -8,56 +8,27 @@
...
 
-.headingItem {
-  flex: 0 14.2857%;
-  color: #444;
-  text-align: right;
-  padding-right: 8px;
-  border-bottom: 2px solid var(--color-mono-gray-4);
+.grid {
+  display: grid;
+  grid-template-columns: repeat(7, 14.2857%);
```
[the remark this answers is not in the document]
Comment (haukkagu) [reply app/routes/events/components/Calendar.css:20]: Will do!

File: app/routes/events/utils.ts (hunk 1 above)
Comment (haukkagu) [reply app/routes/events/utils.ts:51]: Nope, you're right.

File: app/routes/events/components/CalendarCell.tsx [hunk 3]
```diff
@@ -104,13 +112,21 @@ const selectEvents = createSelector(
...
-      .filter((event) => moment(event.startTime).isSame(day, 'day'))
+      .filter(
+        (event) =>
+          moment(event.startTime).isSame(day, 'day') ||
+          (moment(event.startTime).isSameOrBefore(day, 'day') &&
+            moment(event.endTime).isSameOrAfter(day, 'day') &&
+            moment.duration(
+              moment(event.endTime).diff(moment(event.startTime))
+            ) > moment.duration(1, 'days'))
+      )
```
Comment (LudvigHz) [inline app/routes/events/components/CalendarCell.tsx:123]: Is it possible to split this up a bit? Or at least add a comment explaining it :sweat_smile:

File: app/routes/events/utils.ts [hunk 4]
```diff
@@ -47,6 +47,24 @@ export const colorForEvent = (eventType: EventType) => {
...
+  lunch_presentation: '#000',
+  alternative_presentation: '#FFF',
+  course: '#000',
+  breakfast_talk: '#000',
+  party: '#000',
+  social: '#FFF',
+  event: '#FFF',
+  kid_event: 'var(--color-white)',
+  other: 'var(--color-white)',
+};
```
Comment (LudvigHz) [inline app/routes/events/utils.ts:62]: if you slap an `as const` here you should get stronger type checking :)

**Review by LudvigHz — APPROVED** [review]
Awesome :+1: 
Just some small comments, otherwise it looks good to go!

**Review by LudvigHz — APPROVED** [review]
LGTM :shipit:

**Comment by LudvigHz** [comment]
@haukkagu Just fix lint and we can get this in :) And also change your last commit message to something more descriptive at the same time :grimacing:

**Review by ivarnakken — APPROVED** [review]
Looks perfect now!

But like @LudvigHz said, please make the commit message more descriptive ❤️

**Comment by haukkagu** [comment]
Isn't it a good idea to squash the commits before merging? Doesn't that make the name of the last commit irrelevant?
Also, the linting fails on code I haven't touched. What should I? @LudvigHz @ivarnakken

**Pull request merged**